src/Linfo/OS/Linux.php

```diff
@@ -562,6 +562,40 @@ public function getTemps()
                 $return = array_merge($return, $hwmon_vals);
             }
         }
+        
+        // thermal_zone? 
+        if (array_key_exists('thermal_zone', (array) $this->settings['temps']) && !empty($this->settings['temps']['thermal_zone'])) {
+
+            // Store them here
+            $thermal_zone_vals = array();
+
+            // Wacky location
+            foreach ((array) @glob('/sys/class/thermal/thermal_zone*', GLOB_NOSORT | GLOB_BRACE) as $path) {
+                $labelpath = $path.DIRECTORY_SEPARATOR.'type';
+                $valuepath = $path.DIRECTORY_SEPARATOR.'temp';
+
+                // Temperatures
+                if (is_file($labelpath)) {
+                    $label = Common::getContents($labelpath);
+                    $value = Common::getIntFromFile($valuepath);
+                    $value /= $value > 10000 ? 1000 : 1;
+                    $unit = 'C'; // I don't think this is ever going to be in F
+                }
+
+                // Append values
+                $thermal_zone_vals[] = array(
```

Comment:
Can you move the $thermal_zone_vals append to inside the is_file block above? That way if the file doesn't exist we won't be appending missing variables.

Reply:
okay